修改了英文API文档

Atlantisming opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
Others

### PR changes
Docs

### Describe
修改的API文档有：
- paddle.nn.dynamic_decode
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.einsum
- paddle.unique_consecutive
- paddle.disable_signal_handler
- paddle.linalg.svd
- paddle.cumprod
- paddle.device.cuda.stream_guard
- paddle.incubate.segment_min
- paddle.incubate.segment_max
- paddle.incubate.segment_sum
- paddle.incubate.segment_mean

对应的中文修复文档为：[PaddlePaddle/docs/#5429](https://github.com/PaddlePaddle/docs/pull/5429)

Atlantisming committed:
修改paddle.nn.dynamic_decode，paddle.nn.functional.diag_embed 示例

Atlantisming committed:
Merge branch 'PaddlePaddle:develop' into develop

Atlantisming committed:
Merge branch 'PaddlePaddle:develop' into develop

carryyu committed:
mma qk tensor_core (#48087)

* use mma for QK dot computing in fused_multi_transformer.
* Update fused_multi_transformer_op.cu.h

Vvsmile committed:
remove lrn which is not used in paddle 2.0 (#47945)

Vvsmile committed:
replace scatter_nd and scatter_nd_add with paddle.scatter_nd and (#47960)

paddle.scatter_nd_add

Silv3S committed:
[PHI] Migrate mul_grad kernel (#48061)

* cleanup unused code

* unify is_int8 is_bfloat16

* Simplify matmul_v2 FWD kernel

* remove RunKernel methods

* remove import namespace

* remove headers

* clean fluid/phi cross imports

* remove fluid axpy_handler

* delete fluid methods

* activations

* OneDNNMemDesc

* MKLDNNFormatForSize

* MatchShapeToLayout

* MKLDNNMemoryFormat

* MKLDNNFormat

* ReorderMKLDNNHandler

* to_void_cast

* review suggestions

* interpolate

* remove fluid depedency

* init

* ExecuteMatMulV2

* rm fluid kernel

* matmul_grad

* remove mutable_data

* mul_grad

RichardWooSJTU committed:
delete unnecessary shape and slice op (#48112)

Atlantisming committed:
Merge branch 'PaddlePaddle:develop' into develop

Atlantisming committed:
Merge remote-tracking branch 'origin/develop' into my-branch

Atlantisming committed:
修改英文文档。

Atlantisming committed:
Merge branch 'PaddlePaddle:develop' into develop

Atlantisming committed:
Merge branch 'develop' into my-branch

SigureMo commented on python/paddle/incubate/tensor/math.py:
```diff
@@ -45,7 +45,7 @@ def segment_sum(data, segment_ids, name=None):
 
     Returns:
-       output (Tensor): the reduced result.
+       Tensor: the reduced result.
```
这里应该是 `,`，`:` 的话会解析成两行

下同

Atlantisming replied:
已修改

SigureMo commented on python/paddle/incubate/tensor/math.py:
```diff
@@ -28,7 +28,7 @@
 )
 def segment_sum(data, segment_ids, name=None):
-    r"""
+    """
```
带公式的 docstring 是建议加 `r` 的，这样可以保证中英文文档里的公式是一致的，不需要在英文里写双倍的 `\` 来转义，可参考 https://github.com/PaddlePaddle/Paddle/pull/46752#issuecomment-1275551496

所以，只需要移除下面公式里重复的 `\` 即可，如 `$out_i = \\sum_{j} data_{j}$` -> `$out_i = \sum_{j} data_{j}$`

不带公式的且无需转义的可以考虑移除 `r`

下同

Atlantisming replied:
已修改

Atlantisming replied:
像官方书写文档[https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#id6](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#id6)
里面英文模板一样的公式写法
.. math::
           out = x+y
这种格式的文档是不是也移除 `r`

SigureMo replied:
唔，是说多行公式？多行公式和行内公式一样，其实问题不是带不带公式，而是带不带转义字符 `\`，只不过大多数带公式的 docstring 一般都会包含 `\`。

> 这种格式的文档是不是也移除 `r`

为啥要移除呀 :joy:，带公式（或者说带转义）应该加上呀，移除里面重复的 `\`

Atlantisming replied:
噢噢好的，所以官方文档里的公式因为不包含转义符号所以就没有定义成原始字符串。
我不太清楚英文文档生成的语法规则。

Atlantisming replied:
似乎行内公式在没有移除`r`的情况下，只有一个转义字符`\`的话会导致这样的结果
> <img alt="image" width="918" src="https://user-images.githubusercontent.com/39876205/203982107-00d6839e-fb17-4015-b5bd-644e066dfd91.png">
> 
查询了英文文档的生成方式之后，找不到合适的公式表达
> 公式显示奇奇怪怪的，能不能像中文那样公式单独一行居中？

猜测是带`r`的注释，在不转义的情况下是直接将整段字符串传给sphinx，再进行一次编译？
sphinx中的`\`仍为一个转义符号？

SigureMo replied:
直接从中文里 copy 过来即可，我已经修改完成

Atlantisming committed:
修改segment operator等英文文档。

Atlantisming committed:
重新修改了paddle.einsum，paddle.unique_consecutive，
paddle.disable_signal_handler的英文文档格式。

Ligoml commented on python/paddle/nn/functional/extension.py:
```diff
@@ -57,48 +57,48 @@ def diag_embed(input, offset=0, dim1=-2, dim2=-1):
     Examples:
```
api描述的地方把 This OP 这类错误描述去掉吧

Ligoml commented on python/paddle/fluid/layers/rnn.py:
```diff
@@ -1824,7 +1824,6 @@ def dynamic_decode(
 
```
返回值直接罗列吧，说返回一个tuple好像没啥意义，可以参考这个：[paddle.geometric.reindex_graph](https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/geometric/reindex_graph_en.html)

Ligoml commented on python/paddle/tensor/manipulation.py:
```diff
@@ -2296,7 +2296,7 @@ def unique_consecutive(
 
     Returns:
-        tuple (out, inverse, counts). `out` is the unique consecutive tensor for `x`. `inverse` is provided only if `return_inverse` is True. `counts` is provided only if `return_counts` is True.
+        tuple(out, inverse, counts): `out` is the unique consecutive tensor for `x`. `inverse` is provided only if `return_inverse` is True. `counts` is provided only if `return_counts` is True.
```
同 paddle.nn.dynamic_decode 修改建议

Ligoml replied:
note中的 paddle.unique 需要增加一个超链接，方法参考：[飞桨文档相互引用](https://github.com/PaddlePaddle/docs/wiki/%E9%A3%9E%E6%A1%A8%E6%96%87%E6%A1%A3%E7%9B%B8%E4%BA%92%E5%BC%95%E7%94%A8)

Ligoml replied:
似乎还没改完？

Ligoml commented on python/paddle/tensor/linalg.py:
```diff
@@ -2047,7 +2047,7 @@ def svd(x, full_matrices=False, name=None):
             For more information, please refer to :ref:`api_guide_Name`.
```
full_matrices 有默认值，是可选参数，需要注明optional，并说明默认值是什么

Ligoml replied:
中文文档格式有点问题，需要是 ` - **full_matrices** (bool，可选) - 参数描述` 这种格式

Ligoml replied:
tuple改一下吧，中文部分似乎已经改了

Ligoml commented on python/paddle/tensor/math.py:
```diff
@@ -3437,7 +3437,7 @@ def cumprod(x, dim=None, dtype=None, name=None):
 
     Returns:
-        Tensor, the result of cumprod operator.
+        Tensor: the result of cumprod operator.
```
用逗号吧，这个的问题在于 dim 是一个可选参数，中英文文档都需要改

Ligoml commented on python/paddle/device/cuda/__init__.py:
```diff
@@ -355,8 +355,8 @@ def _set_current_stream(stream):
-    **Notes**:
-        **This API only supports dygraph mode currently.**
+    Notes:
+        This API only supports dygraph mode currently.
```
英文文档改对了，中文文档也需要补充哦

Ligoml replied:
```suggestion
        This API only supports dynamic graph mode currently.
```

Ligoml commented on python/paddle/incubate/tensor/math.py:
```diff
@@ -163,7 +163,7 @@ def segment_min(data, segment_ids, name=None):
     This operator calculate the minimum elements of input `data` which with
     the same index in `segment_ids`.
-    It computes a tensor such that $out_i = \\min_{j} data_{j}$
+    It computes a tensor such that $out_i = \min_{j\in \{segment\_ids_j==i\}} data_{j}$
```
<img width="918" alt="image" src="https://user-images.githubusercontent.com/39876205/203982107-00d6839e-fb17-4015-b5bd-644e066dfd91.png">

公式显示奇奇怪怪的，能不能像中文那样公式单独一行居中？

Ligoml replied:
另外中文部分，张量直接写作 Tensor 即可，翻译过来反而增加理解上的困难

Ligoml commented:
顺便说一下，你现在可以在commit名称后面加上 `; test=docs_preview` 来触发文档预览的CI，这样就可以根据预览效果调整啦~参考这个 https://github.com/PaddlePaddle/Paddle/pull/48281

Atlantisming committed:
重新修改了英文文档格式。;test=docs_preview

Ligoml committed:
Update extension.py

Ligoml commented on python/paddle/tensor/linalg.py:
```diff
@@ -2038,16 +2038,21 @@ def svd(x, full_matrices=False, name=None):
+        - S(Tensor), is the singular value decomposition result S.
+
+        - VH(Tensor), VH is the conjugate transpose of V, which is the singular value decomposition result V.
+        Tuple of 3 tensors(U, S, VH): VH is the conjugate transpose of V. S is the singlar value vectors of matrics with shape `[..., K]`
```
<img width="832" alt="image" src="https://user-images.githubusercontent.com/39876205/204492346-25b750d7-7c00-4ceb-9c64-a0d8c9a39dff.png">

报了个错，看上去需要在 Tuple of 3 tensors 之前增加一个空行

Atlantisming committed:
重新修改了英文文档格式。;test=docs_preview

Atlantisming committed:
重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard

待修改：
- paddle.nn.dynamic_decode
- paddle.einsum
- paddle.unique_consecutive
- paddle.linalg.svd
- paddle.uncubate.segment_min
- paddle.uncubate.segment_max
- paddle.uncubate.segment_sum
- paddle.uncubate.segment_mean

;test=docs_preview

Atlantisming committed:
Merge branch 'PaddlePaddle:develop' into my-branch

Atlantisming committed:
重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard
- paddle.nn.dynamic_decode
- paddle.unique_consecutive
- paddle.linalg.svd

待修改：
- paddle.einsum
- paddle.incubate.segment_min
- paddle.incubate.segment_max
- paddle.incubate.segment_sum
- paddle.incubate.segment_mean

;test=docs_preview

Atlantisming committed:
重新修改了英文文档格式。
待验收：
- paddle.linalg.svd
- paddle.nn.functional.diag_embed
- paddle.set_grad_enabled
- paddle.disable_signal_handler
- paddle.cumprod
- paddle.devaice.cuda.stream_guard
- paddle.nn.dynamic_decode
- paddle.unique_consecutive
- paddle.linalg.svd

待修改：
- paddle.einsum
- paddle.incubate.segment_min
- paddle.incubate.segment_max
- paddle.incubate.segment_sum
- paddle.incubate.segment_mean

;test=docs_preview

SigureMo committed:
update

SigureMo committed:
test=docs_preview

SigureMo committed:
update formula; test=docs_preview

SigureMo committed:
update formula; test=docs_preview

SigureMo committed:
remove this operator; test=docs_preview

SigureMo committed:
add hyper link; test=docs_preview

SigureMo committed:
add default value; test=docs_preview

SigureMo committed:
update format; test=docs_preview

SigureMo committed:
empty commit; test=docs_preview

SigureMo commented:
英文里的问题基本都改了下，不过没看预览不保证都对，中文那边就不帮忙改了

SigureMo committed:
fix codestyle issues; test=docs_preview

SigureMo committed:
empty commit; test=docs_preview

Ligoml commented:
<img width="518" alt="image" src="https://user-images.githubusercontent.com/39876205/206125780-ffcf1602-6446-4ec2-8362-a2c0ec2193c6.png">

中文文档这里多了个括号，需要改一下 @Atlantisming

Ligoml reviewed (approved):
good job！LGTM for docs
